File: library/src/main/java/cn/wandersnail/http/upload/UploadService.java
package cn.wandersnail.http.upload;

import java.util.Map;

import okhttp3.MultipartBody;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.HeaderMap;
import retrofit2.http.POST;
import retrofit2.http.Url;

/**
 * date: 2019/8/23 18:09
 * author: zengfansheng
 */
interface UploadService {
    @POST
    Call<ResponseBody> upload(@Url String url, @Body MultipartBody body);

    @POST
    Call<ResponseBody> upload(@Url String url, @Body MultipartBody body, @HeaderMap Map<String, String> headers);
}

File: library/src/main/java/com/snail/network/upload/UploadService.kt
package com.snail.network.upload

import io.reactivex.Observable
import okhttp3.MultipartBody
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.Url

/**
 *
 *
 * date: 2019/2/28 13:48
 * author: zengfansheng
 */
internal interface UploadService {
    @POST
    fun upload(@Url url: String, @Body body: MultipartBody): Observable<Response<ResponseBody>>

    @POST
    fun uploadSync(@Url url: String, @Body body: MultipartBody): Call<ResponseBody>
}